web/src/app/lists/FlatList.tsx

```diff
@@ -162,8 +181,20 @@ export default function FlatList({
   items,
   inset,
   transition,
+  isLoading,
   ...listProps
 }: FlatListProps): JSX.Element {
+  const [displayLoading, setDisplayLoading] = useState(false)
+
+  useEffect(() => {
+    // show loading skeletons after 1 second to avoid load flickering
+    const t = setTimeout(() => {
+      setDisplayLoading(true)
+    }, 1000)
+
+    return () => clearTimeout(t)
+  }, [])
```

Comment:
refer to `Spinner.tsx`'s useEffect.

you could make this a helper function that both components import